staging/src/k8s.io/client-go/util/workqueue/metrics.go

```diff
@@ -210,37 +210,24 @@ func (_ noopMetricsProvider) NewRetriesMetric(name string) CounterMetric {
 	return noopMetric{}
 }
 
-var globalMetricsFactory = queueMetricsFactory{
-	metricsProvider: noopMetricsProvider{},
-}
-
-type queueMetricsFactory struct {
-	metricsProvider MetricsProvider
+var globalMetricsProvider MetricsProvider = noopMetricsProvider{}
 
-	onlyOnce sync.Once
-}
+var setGlobalMetricsProviderOnce sync.Once
```

Comment:
this part is contentious https://github.com/kubernetes/kubernetes/issues/127739

cc: @alvaroaleman @dgrisonnet

Reply:
@alvaroaleman is correct. The change in "style" (removal of the `queueMetricsFactory` factory) is because a struct cannot declare type parameters (generics) that are specific to the method. So we cannot have a non-generic factory struct (necessary to declare a global variable) with methods that use generics.